include/numerics/type_vector.h

```diff
@@ -1168,6 +1168,24 @@ struct CompareTypes<TypeVector<T>, TypeVector<T2>>
 {
   typedef TypeVector<typename CompareTypes<T,T2>::supertype> supertype;
 };
+
+template <typename T, typename T2, typename std::enable_if<ScalarTraits<T>::value, int>::type = 0>
+TypeVector<typename CompareTypes<T, T2>::supertype>
+outer_product(const T & a, const TypeVector<T2> & b)
+{
+  TypeVector<typename CompareTypes<T, T2>::supertype> ret;
+  for (unsigned int i = 0; i < LIBMESH_DIM; i++)
+    ret(i) = a * b(i);
+
+  return ret;
+}
+
+template <typename T, typename T2, typename std::enable_if<ScalarTraits<T2>::value, int>::type = 0>
+TypeVector<typename CompareTypes<T, T2>::supertype>
+outer_product(const TypeVector<T> & a, const T2 & b)
+{
+  return outer_product(b, a);
+}
```

Comment:
I feel like if I point out "this isn't correct for non-commutative `*`" then I'm being an ass, but if I don't point out "this isn't correct for non-commutative `*`" then karma will determine that Now Is The Time for someone to try and start stuffing quaternions or something into a `TypeVector`.